src/Components/WebAssembly/Server/src/ComponentsWebAssemblyEndpointRouteBuilderExtensions.cs

```diff
@@ -97,6 +102,22 @@ private static StaticFileOptions CreateStaticFilesOptions(IFileProvider webRootF
 
             options.ContentTypeProvider = contentTypeProvider;
 
+            // Static files middleware will try to use application/x-gzip as the content
+            // type when serving a file with a gz extension. We need to correct that before
+            // sending the file.
+            options.OnPrepareResponse = fileContext =>
+            {
+                var requestPath = fileContext.Context.Request.Path;
+                if (string.Equals(Path.GetExtension(requestPath.Value), ".gz"))
+                {
+                    var originalPath = Path.GetFileNameWithoutExtension(requestPath.Value);
+                    if (contentTypeProvider.TryGetContentType(originalPath, out var originalContentType))
+                    {
+                        fileContext.Context.Response.ContentType = originalContentType;
+                    }
+                }
```

Comment:
This block seems a bit dubious, in that if we're serving a file called `something.tar.gz`, this logic doesn't account for whether the client thinks it's actually getting `something.tar` or `something.tar.gz`. The only code that really knows that is below, inside `NegotiateEncoding`, since that's what actually rewrites `context.Request.Path`.

Would it be worth updating `NegotiateEncoding` so that, when rewriting the request, it stashes some extra info in `context` to say what the original request path was? Then we can just look for that here and unconditionally use the response content type for that original request path.

Reply:
I know this doesn't technically affect us until there's some file in `_framework` that has the `.gz` extension naturally (not due to the compression feature you're adding here). But if we do add such a file one day, we'll get really confused about why it's being served with the wrong content type.

Reply:
Are you OK if we tackle this when we decide to compress other files? I have ideas that will render this obsolete, and is something we plan to change, so I rather tackle the cleanup at that point.

Reply:
Sure. Would it be OK to add a brief comment into this block indicating this intent? Just so we know what's going on if the question comes up again in later work on this.

Reply:
I've added a comment for this.